models/tokens_models.go

```diff
@@ -0,0 +1,98 @@
+package models
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/ONSdigital/dp-identity-api/apierrorsdeprecated"
+	"github.com/ONSdigital/dp-identity-api/utilities"
+	"github.com/ONSdigital/log.go/log"
+	"github.com/aws/aws-sdk-go/service/cognitoidentityprovider"
+	"github.com/dgrijalva/jwt-go"
+)
+
+type IdClaims struct {
+	Sub           string `json:"sub"`
+	Aud           string `json:"aud"`
+	EmailVerified bool   `json:"email_verified"`
+	TokenUse      string `json:"token_use"`
+	AuthTime      int    `json:"auth_time"`
+	Iss           string `json:"iss"`
+	CognitoUser   string `json:"cognito:username"`
+	Exp           int    `json:"exp"`
+	GivenName     string `json:"given_name"`
+	Iat           int    `json:"iat"`
+	Email         string `json:"email"`
+	jwt.StandardClaims
+}
+
+type IdToken struct {
+	TokenString string
+	Claims      IdClaims
+}
+
+func (t *IdToken) ParseWithoutValidating(tokenString string) error {
+	parser := new(jwt.Parser)
+
+	idToken, _, parsingErr := parser.ParseUnverified(tokenString, &IdClaims{})
+	if parsingErr != nil {
+		fmt.Println(parsingErr)
+		return parsingErr
+	}
+
+	idClaims, ok := idToken.Claims.(*IdClaims)
+	if !ok {
+		return errors.New("the ID token could not be parsed")
+	}
+
+	t.Claims = *idClaims
+	return nil
+}
+
+func (t *IdToken) Validate(ctx context.Context, errorList *[]apierrorsdeprecated.Error) {
+	if t.TokenString == "" {
+		invalidIDTokenErrorBody := apierrorsdeprecated.IndividualErrorBuilder(apierrorsdeprecated.InvalidIDTokenError,
+			apierrorsdeprecated.MissingIDTokenMessage)
+		*errorList = append(*errorList, invalidIDTokenErrorBody)
+		log.Event(ctx, apierrorsdeprecated.MissingRefreshTokenMessage, log.ERROR)
+	} else {
```

Comment:
We can avoid the else if we return from the end of the 'if' block. Then all the contained code moves back to the primary line of sight / happy path

Reply:
Updated to as described